File: packages/sdk-csharp/src/TinyHuman.Sdk/TinyHumanError.cs
namespace TinyHuman.Sdk;

public class TinyHumanError : Exception
{
    public int Status { get; }
    public string Body { get; }

    public TinyHumanError(string message, int status, string body)
        : base(message)
    {
        Status = status;
        Body = body;
    }
}

File: packages/sdk-dart/lib/src/tinyhuman_error.dart
class TinyHumanError implements Exception {
  final String message;
  final int status;
  final String body;

  TinyHumanError(this.message, this.status, [this.body = '']);

  @override
  String toString() => 'TinyHumanError($status): $message';
}
